buildtools/npm/lockfile.go

```diff
@@ -0,0 +1,125 @@
+package npm
+
+import (
+	"path/filepath"
+
+	"github.com/fossas/fossa-cli/errors"
+	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+var PossibleLockfileFilenames = [...]string{"package-lock.json", "npm-shrinkwrap.json"}
+
+type Lockfile struct {
+	Name         string
+	Version      string
+	Dependencies Dependencies
+}
+
+type DependencyLockEntry struct {
+	Version      string
+	Requires     map[string]string
+	Dependencies map[string]*DependencyLockEntry
```

Comment:
You should be able to fill in the type you defined on line 26 here so that it is `Dependencies Dependencies`

Reply:
I didn't know you could have circular types like that. That's cool!